src/components/Map/index.tsx

```diff
@@ -169,28 +175,54 @@ const Map = () => {
     };
   }, [airQualityByCity]);
 
+  const handleClickMarker = useCallback((city: HTMLDivElement) => {
+    setCity(city.id);
+    if (city.dataset.finedustgrade) {
+      setFineDustScale(+city.dataset.finedustgrade);
+    }
+    if (city.dataset.ultrafinedustgrade) {
+      setUltraFineDustScale(+city.dataset.ultrafinedustgrade);
+    }
+    onOpen();
+  }, []);
```

Comment:
```suggestion
  const handleClickMarker = useCallback((city: HTMLDivElement) => {
    setCity(city.id);
    
    const grade = city.dataset;
    if(!grade) return;

    if (grade.finedustgrade) {
      setFineDustScale(Number(city.dataset.finedustgrade));
    }
    if (grade.ultrafinedustgrade) {
      setUltraFineDustScale(Number(city.dataset.ultrafinedustgrade));
    }
    
    onOpen();
  }, []);
```

이렇게로도 작성할 수 있을 것 같아요!